Add uwsgi_graphite_extraopts to configure uwsgi

mtekel opened the pull request:
Defining this variable you can add any custom uwsgi configuration to uwsgi_graphite.ini, such as e.g. enabling http on port 8080 or enabling http auth:

```
uwsgi_graphite_extraopts: |
  http = {{ ansible_default_ipv4.address }}:8080
  plugins = router_basicauth
  route = ^/ basicauth:myRealm,foo:bar
```

nsg commented:
Not sure about the use of `|`, not that many people know about that part of the yaml definition. I think a list like this would be cleaner:

``` yaml
uwsgi_graphite_extraopts:
  - option: http
    value: "{{ ansible_default_ipv4.address }}:8080"
  - option: plugins
    value: router_basicauth
  - option: route
    value: "^/ basicauth:myRealm,foo:bar"
```

... and something like this:

```
{% for opts in uwsgi_graphite_extraopts %}
{{ opts.option }}={{ opts.value }}
{% endfor %}
```

What do you think? I do not like the "everything in a huge string" approach :)

mtekel committed:
Add uwsgi_graphite_extraopts to configure uwsgi

Defining this variable you can add any custom uwsgi configuration to uwsgi_graphite.ini, such as e.g. enabling http on port 8080 or enabling http auth:
```
uwsgi_graphite_extraopts:
  - option: http
    value: "{{ ansible_default_ipv4.address }}:8080"
  - option: plugins
    value: router_basicauth
  - option: route
    value: "^/ basicauth:myRealm,foo:bar"
```

mtekel commented:
Done, but I haven't tested it yet.

nsg commented:
I think you also need to add `uwsgi_graphite_extraopts: []` to defaults.

mtekel commented:
Done. Tried to test it today, but I got `ImportError: No module named fields` on syncDB, which is similar to this issue: https://bugs.launchpad.net/graphite/+bug/870078 . I suspect it has something to do with combination of python, pip and the specific ubuntu image I used...

nsg commented:
I fixed that in https://github.com/nsg/ansible-graphite/commit/a648f21571e41b62d471e4bf93bc6b58c8b51462, it looks like django-tagging has dropped support for the django version that we are using.

nsg commented:
Sorry for leaving this PR here open for this long, I will merge it now. :+1:

mtekel commented:
Thanks, no harm done in delay as I wasn't depending on this.
